src/api/objects/InMemorySearchProvider.js

```diff
@@ -317,26 +379,87 @@ class InMemorySearchProvider {
      * @private
      * @returns {String} a unique query Id for the query.
      */
-    dispatchSearch(queryId, searchInput, maxResults) {
+    dispatchSearch(queryId, searchType, searchInput, maxResults) {
         const message = {
-            request: 'search',
+            request: searchType,
             input: searchInput,
             maxResults,
             queryId
         };
         this.worker.port.postMessage(message);
     }
 
+    localIndexTags(keyString, objectToIndex, model) {
```

Comment:
adding tags that don't exist in the old model, then removing tags that aren't present in the new model.